pc_hal/conioHistory.hpp

```diff
@@ -0,0 +1,280 @@
+         * @param _bottom_display New bottom display index
+         */
+        void set_bottom_display(std::size_t _bottom_display);
+
+        /**
+         * @brief Get the current bottom display index
+         *
+         * @return Current bottom display index
+         */
+        std::size_t get_cur_bottom_display() const;
+
+        /**
+         * @brief Set the current bottom display index
+         *
+         * @param _cur_bottom_display New current bottom display index
+         */
+        void set_cur_bottom_display(std::size_t _cur_bottom_display);
+
+        /**
+         * @brief Get the current status of the history logging
+         * 
+         * @return true if active, false if not
+         */
+        bool is_active() const;
+
+    private:
+        /**
+         * @brief Struct that contains the necessary metadata for a line in the history
+         * 
+         */
+        typedef struct CONIO_history_line_
+        {
+            /**
+             * @brief Starting offset within the file
+             * 
+             */
+            std::size_t offset;
+            /**
+             * @brief Length of the line
+             * 
+             */
+            std::size_t len;
+            /**
+             * @brief Fragment sequence number within the line
+             * 
+             */
+            std::uint32_t frag_seq;
+            /**
+             * @brief Display flag: false = do not display, true = display to window
+             *
+             */
+            bool display;
+            /**
+             * @brief More Framgents flag: false = no successive fragments, true = more fragments for the line
+             *
+             */
+            bool more_frag;
+            // Additional metadata may be necessary
+
+            /**
+             * @brief Construct a new conio history line object with manually set parameters
+             * 
+             * @param _offset Starting offset
+             * @param _len Current length of the line
+             * @param _display Display flag
+             * @param _more_frag More Fragments flag
+             * @param _frag_seq Fragment sequence number
+             */
+            CONIO_history_line_(std::size_t _offset,
+                                std::size_t _len,
+                                std::uint32_t _frag_seq,
+                                bool _display,
+                                bool _more_frag);
+            /**
+             * @brief Construct a new conio history line object with default flags
+             *
+             * @param _offset Starting offset
+             */
+            CONIO_history_line_(std::size_t _offset);
+        } CONIO_hist_line;
+
+        /**
+         * @brief Flag to indicate if the history logging is active
+         * @details This is flag is only set to false if there is an error with mapping the file and should have deinit_file_mapping be called to do so.
+         */
+        bool active = true;
```

Comment:
Setting this true here will lead to incorrect behavior, as calls prior to `init_file_mapping` will utilize the incorrect state.